src/qspdlog_model.cpp

```diff
@@ -145,3 +160,39 @@ QVariant QSpdLogModel::headerData(
 
     return QVariant();
 }
+void QSpdLogModel::setForegroundMapping(
+    const std::string& loggerName, std::optional<QBrush> brush
+)
+{
+    if(brush.has_value())
```

Comment:
You could also use the following notation with the `std::optional`, though it's also fine, and maybe much better the way you wrote.
```suggestion
    if(brush)
```